lib/ui/painting.dart

```diff
@@ -1174,13 +1174,31 @@ class Vertices extends NativeFieldWrapperClass2 {
       throw new ArgumentError("[indices] values must be valid indices in the positions list");
 
     Float32List encodedPositions = _encodePointList(positions);
-    Float32List encodedTexs = (textureCoordinates != null) ?
+    Float32List encodedTextureCoordinates = (textureCoordinates != null) ?
       _encodePointList(textureCoordinates) : null;
     Int32List encodedColors = (colors != null) ? _encodeColorList(colors) : null;
     Int32List encodedIndices = (indices != null) ? new Int32List.fromList(indices) : null;
 
     _constructor();
-    _init(mode.index, encodedPositions, encodedTexs, encodedColors, encodedIndices);
+    _init(mode.index, encodedPositions, encodedTextureCoordinates, encodedColors, encodedIndices);
+  }
+
+  Vertices.raw(
+    VertexMode mode,
+    Float32List positions, {
+    Float32List textureCoordinates,
+    Int32List colors,
+    Int32List indices,
+  }) {
+    if (textureCoordinates != null && textureCoordinates.length != positions.length)
+      throw new ArgumentError("[positions] and [textureCoordinates] lengths must match");
```

Comment:
there's several examples of these, please audit all calls to `new ArgumentError`